langchain/src/schema/runnable/remote.ts

```diff
@@ -0,0 +1,312 @@
+          [...pk, k],
+        ] as [
+          RunnableConfig[],
+          Omit<Partial<CallOptions>, keyof BaseCallbackConfig>[]
+        ],
+      [[], []] as [
+        RunnableConfig[],
+        Omit<Partial<CallOptions>, keyof BaseCallbackConfig>[]
+      ]
+    );
+    const [configs, kwargs] = tup ?? [undefined, undefined];
+    const response = await this.post<{
+      inputs: RunInput[];
+      config?: (RunnableConfig & RunnableBatchOptions)[];
+      kwargs?: Omit<Partial<CallOptions>, keyof BaseCallbackConfig>[];
+    }>("/batch", {
+      inputs,
+      config: (configs ?? [])
+        .map(withoutCallbacks)
+        .map((config) => ({ ...config, ...batchOptions })),
+      kwargs,
+    });
+    const body = await response.json();
+
+    if (!body.output) throw new Error("Invalid response from remote runnable");
+
+    return revive(body.output);
+  }
+
+  async batch(
+    inputs: RunInput[],
+    options?: Partial<RunnableConfig> | Partial<RunnableConfig>[],
+    batchOptions?: RunnableBatchOptions & { returnExceptions?: false }
+  ): Promise<RunOutput[]>;
+
+  async batch(
+    inputs: RunInput[],
+    options?: Partial<RunnableConfig> | Partial<RunnableConfig>[],
+    batchOptions?: RunnableBatchOptions & { returnExceptions: true }
+  ): Promise<(RunOutput | Error)[]>;
+
+  async batch(
+    inputs: RunInput[],
+    options?: Partial<RunnableConfig> | Partial<RunnableConfig>[],
+    batchOptions?: RunnableBatchOptions
+  ): Promise<(RunOutput | Error)[]>;
+
+  async batch(
+    inputs: RunInput[],
+    options?: Partial<RunnableConfig> | Partial<RunnableConfig>[],
+    batchOptions?: RunnableBatchOptions
+  ): Promise<(RunOutput | Error)[]> {
+    if (batchOptions?.returnExceptions) {
+      throw Error("returnExceptions is not supported for remote clients");
+    }
+    return this._batchWithConfig(
+      this._batch.bind(this),
+      inputs,
+      options,
+      batchOptions
+    );
+  }
+
+  async stream(
+    input: RunInput,
+    options?: RunnableConfig
+  ): Promise<IterableReadableStream<RunOutput>> {
+    const response = await this.post<{
+      input: RunInput;
+      options: RunnableConfig;
+    }>("/stream", {
+      input,
+      options: options ?? {},
+    });
+    if (!response.ok) {
+      const json = await response.json();
+      const error = new Error(
+        `RemoteRunnable call failed with status code ${response.status}: ${json.message}`
+      );
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (error as any).response = response;
+      throw error;
+    }
+    const { body } = response;
+    if (!body) {
+      throw new Error(
+        "Could not begin LangServe stream. Please check the given URL and try again."
+      );
+    }
+    const stream = new ReadableStream({
+      async start(controller) {
+        const enqueueLine = getMessages((msg) => {
+          if (msg.data) controller.enqueue(deserialize(msg.data));
+        });
+        const onLine = (
```

Comment:
`onLine`?

Reply:
Haha that's what it's called in the event-source-parse.ts file. Just a callback for each line. Might be worth reworking some of that logic to use proper streams rather than these callbacks.